### PR TITLE
fix: atomic db state listener insert

### PR DESCRIPTION
**Pull request by MujkicA**
co-authored with @segfault-magnet

### CODE REVIEW

File: f.patch
Comment (digorithm) [inline f.patch:1]: Was this supposed to be committed?

File: packages/services/src/state_listener.rs [hunk 1]
```diff
@@ -89,9 +96,8 @@ where
             {
                 // tx included in block but is not yet finalized
                 if tx.state == TransactionState::Pending {
-                    self.storage
-                        .update_tx_state(tx.hash, TransactionState::IncludedInBlock)
-                        .await?;
+                    // todo dzematile sve ostale na failed a ovu na included in block
```
Comment (digorithm) [inline packages/services/src/state_listener.rs:99]: ?

File: packages/services/src/state_listener.rs [hunk 2]
```diff
@@ -40,17 +42,24 @@ where
     async fn check_non_finalized_txs(&mut self, non_finalized_txs: Vec<L1Tx>) -> crate::Result<()> {
         let current_block_number: u64 = self.l1_adapter.get_block_number().await?.into();
 
+        let mut skip_nonces = HashSet::new();
+
+        let mut selective_change = vec![];
+        let mut noncewide_changes = vec![];
```
Comment (digorithm) [inline packages/services/src/state_listener.rs:48]: Probably you guys are gonna do this soon, but just to make sure: can we get some comments here explaining the rationale for these two vecs and the strategy we're using here?

**Review by Br1ght0ne — DISMISSED** [review]
LGTM!

**Review by Br1ght0ne — APPROVED** [review]

**Review by Br1ght0ne — DISMISSED** [review]

**Review by segfault-magnet — APPROVED** [review]

**Review by hal3e — APPROVED** [review]
great work!

**Review by Br1ght0ne — APPROVED** [review]

**Pull request merged**